src/main/java/org/opentripplanner/street/model/edge/StreetTransitStationCentroidLink.java

```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.street.model.edge;
+
+import org.opentripplanner.street.model.vertex.StreetVertex;
+import org.opentripplanner.street.model.vertex.TransitStationCentroidVertex;
+import org.opentripplanner.transit.model.basic.Accessibility;
+
+public class StreetTransitStationCentroidLink
+  extends StreetTransitEntityLink<TransitStationCentroidVertex> {
+
+  private final boolean isEntrance;
+
+  private StreetTransitStationCentroidLink(StreetVertex fromv, TransitStationCentroidVertex tov) {
+    super(fromv, tov, tov.getStation().getWheelchairAccessibility());
+    isEntrance = true;
+  }
+
+  private StreetTransitStationCentroidLink(TransitStationCentroidVertex fromv, StreetVertex tov) {
+    super(fromv, tov, fromv.getStation().getWheelchairAccessibility());
+    isEntrance = false;
+  }
+
+  public static StreetTransitStationCentroidLink createStreetTransitStationLink(
+    StreetVertex fromv,
+    TransitStationCentroidVertex tov
+  ) {
+    return connectToGraph(new StreetTransitStationCentroidLink(fromv, tov));
+  }
+
+  public static StreetTransitStationCentroidLink createStreetTransitStationLink(
+    TransitStationCentroidVertex fromv,
+    StreetVertex tov
+  ) {
+    return connectToGraph(new StreetTransitStationCentroidLink(fromv, tov));
+  }
+
+  public boolean isEntrance() {
+    return isEntrance;
+  }
+
+  public boolean isExit() {
+    return !isEntrance;
+  }
+
+  public String toString() {
+    return "StreetTransitStationCentroidLink(" + fromv + " -> " + tov + ")";
+  }
+
+  protected int getStreetToStopTime() {
+    return 0;
```

Comment:
Is this accurate? Wouldn't that give you the wrong time if you had to walk to a far away platform?

On the other hand it would probably hard to come up with a good formula for that and since it's a rarely used feature it's probably not very important.

Reply:
As I understand it this is a feature for subways to simulate the time it takes to get from the street level to the underground stop? I think even if the stops are underground, you would want the station to be at street level. Like the main entrance or something like that. But I'm not completely sure

Reply:
Very little documentation on the super method, so difficult to tell how this should be implemented.

But, if this is the time to travel from the OSM network to the Station centroid, then 0 is probably ok. We are going to use this to prefer stops closer to the centroid, over stops fare from it. We do not want the passenger to travel here -or provide any walk-legs/directions here. Adding a "line of sight" time might improve the results, but unsure how much.

Reply:
Just to clarify: in general this PR won't prefer child stops closer to the centroid, since if you search from a station id you get direct accesses to all the child stops. Only if you actually have to walk to a stop that is not part of the station will you use the centroid as the starting point.

Regarding the `streetToStopTime` I think we agree that it's ok to leave it at 0 then.